mods/beds/functions.lua

```diff
@@ -224,6 +239,27 @@ minetest.register_on_leaveplayer(function(player)
 	end
 end)
 
+minetest.register_on_dieplayer(function(player)
+	local name = player:get_player_name()
+	local in_bed = beds.player
+	local pos = player:get_pos()
+	local yaw = get_look_yaw(pos)
+
+	if in_bed[name] then
+		lay_down(player, nil, pos, false)
+		player:set_look_horizontal(yaw)
+		player:set_pos(pos)
+	end
+end)
+
+local old_calculate_knockback = minetest.calculate_knockback
+function minetest.calculate_knockback(player, ...)
+	if beds.player[player:get_player_name()] then
+		return 0
+	end
```

Comment:
should be removed since #2620 was merged

Reply:
Done!